src/App.js

```diff
@@ -1,26 +1,114 @@
 import React from 'react';
-import logo from './logo.svg';
+import Bucket from './Bucket';
 import './App.css';
+import { Container } from 'reactstrap';
+import {
+  InputGroup,
+  InputGroupAddon,
+  InputGroupText,
+  Col,
+  Row,
+  Label,
+  Input
+} from 'reactstrap';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+class App extends React.Component {
+  state = {
+    income: 0,
+    buckets: {
```

Comment:
Have you considered making this an array of objects? Something like

```js
buckets: [
  { name: "Fixed Costs", value: 50 },
]
```

Reply:
Could you still update them on the fly like that?